execute/tokendata/usdc/attestation.go

```diff
@@ -59,15 +62,14 @@ type sequentialAttestationClient struct {
 	hasher hashutil.Hasher[[32]byte]
 }
 
-//nolint:revive
-func NewAttestationClient(config pluginconfig.USDCCCTPObserverConfig) (*sequentialAttestationClient, error) {
+func NewSequentialAttestationClient(config pluginconfig.USDCCCTPObserverConfig) (AttestationClient, error) {
 	client, err := NewHTTPClient(
 		config.AttestationAPI,
 		config.AttestationAPIInterval.Duration(),
 		config.AttestationAPITimeout.Duration(),
 	)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create HTTP client: %w", err)
```

Comment:
general tip, no need to say `failed to...` in an error. "create http client: %w" is fine, it's an error anyways